core/frontend/src/IModelApp.ts

```diff
@@ -1,699 +1,704 @@
+import * as categorySelectorState from "./CategorySelectorState";
+import * as displayStyleState from "./DisplayStyleState";
+import * as drawingViewState from "./DrawingViewState";
+import { ElementLocateManager } from "./ElementLocateManager";
+import { EntityState } from "./EntityState";
+import { FrontendHubAccess } from "./FrontendHubAccess";
+import { FrontendLoggerCategory } from "./FrontendLoggerCategory";
+import * as modelselector from "./ModelSelectorState";
+import * as modelState from "./ModelState";
+import { NotificationManager } from "./NotificationManager";
+import { QuantityFormatter } from "./quantity-formatting/QuantityFormatter";
+import { RenderSystem } from "./render/RenderSystem";
+import { System } from "./render/webgl/System";
+import * as sheetState from "./SheetViewState";
+import * as spatialViewState from "./SpatialViewState";
+import { TentativePoint } from "./TentativePoint";
+import { MapLayerFormatRegistry, MapLayerOptions, TileAdmin } from "./tile/internal";
+import * as accudrawTool from "./tools/AccuDrawTool";
+import * as clipViewTool from "./tools/ClipViewTool";
+import * as idleTool from "./tools/IdleTool";
+import * as measureTool from "./tools/MeasureTool";
+import * as selectTool from "./tools/SelectTool";
+import { ToolRegistry } from "./tools/Tool";
+import { ToolAdmin } from "./tools/ToolAdmin";
+import * as viewTool from "./tools/ViewTool";
+import { UserPreferencesAccess } from "./UserPreferences";
+import { ViewManager } from "./ViewManager";
+import * as viewState from "./ViewState";
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+require("./IModeljs-css");
+
+// cSpell:ignore noopener noreferrer gprid forin nbsp csrf xsrf
+
+/** Options that can be supplied with [[IModelAppOptions]] to customize frontend security.
+ * @public
+ */
+export interface FrontendSecurityOptions {
+  /** Configures protection from Cross Site Request Forgery attacks. */
+  readonly csrfProtection?: {
+    /** If enabled, IModelApp will extract the CSRF token for the current session from the document's cookies and send it with each request as a header value. */
+    readonly enabled: boolean;
+    /** Defaults to XSRF-TOKEN. */
+    readonly cookieName?: string;
+    /** Defaults to X-XSRF-TOKEN. */
+    readonly headerName?: string;
+  };
+}
+
+/** Options that can be supplied to [[IModelApp.startup]] to customize frontend behavior.
+ * @public
+ */
+export interface IModelAppOptions {
+  /** If present, supplies the [[IModelClient]] for this session. */
+  hubAccess?: FrontendHubAccess;
+  /** If present, supplies the Id of this application. Applications must set this to the Bentley Global Product Registry Id (GPRID) for usage logging. */
+  applicationId?: string;
+  /** If present, supplies the version of this application. Must be set for usage logging. */
+  applicationVersion?: string;
+  /** If present, supplies the [[UserPreferencesAccess]] for this session.
```

Comment:
You have deleted the `@public` `IModelApp.settings` and replaced with the `@beta` `IModelApp.userPreferences`. Should mention this in NextVersion.md.

Reply:
Added a section